mlrun/datastore/sources.py

```diff
@@ -412,6 +424,92 @@ def to_dataframe(
+            if not filter_tuple:
+                continue
+            col_name, op, value = filter_tuple
+            if op.lower() in ("in", "not in") and isinstance(value, (list, tuple, set)):
+                none_exists = False
+                value = list(value)
+                for none_value in none_values:
+                    if none_value in value:
+                        value.remove(none_value)
+                        none_exists = True
+                if none_exists:
+                    filter_nan = column_types[col_name] != "timestamp"
+                    if value:
+                        if op.lower() == "in":
+                            new_filter = (
+                                col(col_name).isin(value) | col(col_name).isNull()
+                            )
+                            new_filter = (
+                                new_filter | isnan(col(col_name))
+                                if filter_nan
+                                else new_filter
+                            )
+                        else:
+                            new_filter = (
+                                ~col(col_name).isin(value) & ~col(col_name).isNull()
+                            )
+                            new_filter = (
+                                new_filter & ~isnan(col(col_name))
+                                if filter_nan
+                                else new_filter
+                            )
+                    else:
+                        if op.lower() == "in":
+                            new_filter = col(col_name).isNull()
+                            new_filter = (
+                                new_filter & isnan(col(col_name))
+                                if filter_nan
+                                else new_filter
+                            )
+                        else:
+                            new_filter = ~col(col_name).isNull() & ~isnan(col(col_name))
+                            new_filter = (
+                                new_filter & ~isnan(col(col_name))
+                                if filter_nan
+                                else new_filter
+                            )
+                else:
+                    if op.lower() == "in":
+                        new_filter = col(col_name).isin(value)
+                    elif op.lower() == "not in":
+                        new_filter = ~col(col_name).isin(value)
+            else:
+                new_filter = operators[op](col(col_name), value)
+            if spark_filter is not None:
+                spark_filter = spark_filter & new_filter
+            else:
+                spark_filter = new_filter
+        return spark_filter
+
+    def _filter_spark_df(self, df, time_field=None, columns=None):
+        spark_additional_filters = self._get_spark_additional_filters(
+            column_types=dict(df.dtypes)
+        )
+        if spark_additional_filters is not None:
```

Comment:
I think here and in a few other places, it's possible to use the more idiomatic
```suggestion
        if spark_additional_filters:
```
dropping the `is not None`.

Reply:
you can not use if condition with spark Column object.
That is why I have not use not None...